orc8r/gateway/python/magma/common/sentry.py

```diff
@@ -83,30 +85,56 @@ def get_sentry_status(service_name: str) -> SentryStatus:
         return SentryStatus.DISABLED
 
 
-def sentry_init(service_name: str):
-    """Initialize connection and start piping errors to sentry.io."""
+# TODO when control_proxy.yml is outdated move to shared mconfig entirely
+def get_sentry_dsn_and_sample_rate(sentry_mconfig: mconfigs_pb2.SharedSentryConfig) -> Tuple[str, float]:
+    """Get Sentry configs with the following priority:
+    1) control_proxy.yml (if sentry_python_url is present)
+    2) shared mconfig (i.e. first streamed mconfig from orc8r,
+    if not present default mconfig in /etc/magma)
 
-    sentry_status = get_sentry_status(service_name)
-    if sentry_status == SentryStatus.DISABLED:
-        return
+    Args:
+            sentry_mconfig (SharedSentryConfig): proto message
+            type of shared mconfig
+    Returns:
+            (str, float): sentry url, sentry sample rate
+    """
 
-    sentry_url = get_service_config_value(
+    dsn_python = get_service_config_value(
         CONTROL_PROXY,
         SENTRY_URL,
         default='',
     )
-    if not sentry_url:
+
+    if not dsn_python:
+        dsn_python = sentry_mconfig.dsn_python
+        sample_rate = sentry_mconfig.sample_rate
+        return dsn_python, sample_rate
+    else:
+        sample_rate = get_service_config_value(
+            CONTROL_PROXY,
+            SENTRY_SAMPLE_RATE,
+            default=DEFAULT_SAMPLE_RATE,
+        )
+    return dsn_python, sample_rate
+
+
+def sentry_init(service_name: str, sentry_mconfig: mconfigs_pb2.SharedSentryConfig) -> None:
+    """Initialize connection and start piping errors to sentry.io."""
+
+    sentry_status = get_sentry_status(service_name)
+    if sentry_status == SentryStatus.DISABLED:
+        return
+
+    dsn_python, sample_rate = get_sentry_dsn_and_sample_rate(sentry_mconfig)
+
+    if not dsn_python:
+        logging.info('Sentry disabled because of missing dsn_python.')
```

Comment:
Can we add some more detail on where this value can be placed? (So we can say what's described in the function doc above somewhere. control_proxy config for per-agw override and mconfig.shared_config.sentry_config for network wide configs)

Reply:
Hmm actually, maybe it is too noisy if we print that for every service

Reply:
True, it might be... Do you think printing the single line, as we currently do, might be too noisy as well? (On the other hand it may be helpful for users, who wonder why sentry isn't working, even though they enabled it in the service config.)

Reply:
Yeah since we already have the single line message and that doesn't seem to bother us, let's just make it a bit more descriptive then :) You make a good point about there being an additional config about enabling / disabling sentry so it is confusing.